images/router/haproxy/reload-haproxy

```diff
@@ -15,7 +16,37 @@ if [ -f $pid_file ]; then
 fi
 
 if [ -n "$old_pid" ]; then
+  if $(set | grep DROP_SYN_DURING_RESTART= > /dev/null) && [ -n "$DROP_SYN_DURING_RESTART" ]; then
+    # We install the syn eater so that connections that come in during the restart don't
+    # go onto the wrong socket, which is then closed.  The sleep is needed to let the socket
+    # drain before the new haproxy starts and binds to the same port.
+    ports=$(grep 'bind :' $config_file | cut -f2 -d: | paste -d, -s)
+
+    if [ -n "$ports" ]; then
+      # If this doesn't insert, we don't care, we still want to reload
+      /usr/sbin/iptables -I INPUT -p tcp -m multiport --dports $ports --syn -j DROP \
+			 -m comment --comment "Eat SYNs while reloading haproxy"
```

Comment:
`|| :`  - ignore error if we don't care - just in case we ever put a `set -e` in.